src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
***Required**
1. Looks simple and understandable but in this case, will be better to use reduce.
2. Always try to use a map instead forEach to avoid mutation. And the map works faster.
[Iterators and Generators](https://mate-academy.github.io/style-guides/javascript.html#iterators-and-generators)
```
  const menAgeSum = men.reduce((sumAges, person) => {
    return sumAges + (person.died - person.born) / menCentury.length;
  }, 0);
```